File: BoardGameNight/ClientApp/src/app/session.ts
import { GameShelf } from "./game-shelf";
import { GameNightEvent } from "./gamenightevent";

export interface Session {
    id: number;
    timePlayed: number;
    winner: string;
    enjoyment: number;
    ownedId: number;
    event:GameNightEvent;
    owned: GameShelf;
    events: GameNightEvent[];
    sessionAttendees: any[];
}

File: BoardGameNight/Models/Session.cs
using System;
using System.Collections.Generic;

namespace BoardGameNight.Models
{
    public partial class Session
    {
        public Session()
        {
            Events = new HashSet<Event>();
            SessionAttendees = new HashSet<SessionAttendee>();
        }

        public int Id { get; set; }
        public double? TimePlayed { get; set; }
        public string? Winner { get; set; }
        public int? Enjoyment { get; set; }
        public int? OwnedId { get; set; }
        public int? EventId { get; set; }

        public virtual Event? Event { get; set; }
        public virtual GameShelf? Owned { get; set; }
        public virtual ICollection<Event> Events { get; set; }
        public virtual ICollection<SessionAttendee> SessionAttendees { get; set; }
    }
}
